### ISSUE TITLE
Enhancement: calcite-input type=number - only allow integer inputs

### ISSUE THREAD

**Issue by AdelheidF** [Esri/calcite-components#2009]
### Description

We have places where we use calcite-input and only allow integer values. It would be nice if calcite-input took care of this and not allow decimal separators. Otherwise we have to check the value in the event and 'fix' it accordingly everywhere we use it.

It would be great if it would stop you from doing that, even though once you hit enter it becomes a rounded number again.
![image](https://user-images.githubusercontent.com/44481292/115594986-3ba8bc80-a28b-11eb-822d-6b1a89b8b8b6.png)

**Comment by jcfranco** [comment]
Verified with https://codepen.io/jcfranco/pen/qBmYONO

@eriklharper Please avoid closing issues without verification.

**Comment by eriklharper** [comment]
> Verified with https://codepen.io/jcfranco/pen/qBmYONO
> 
> @eriklharper Please avoid closing issues without verification.

Did I miss the memo on that?  Should we be creating codepens for every resolved issue now?

**Comment by AdelheidF** [comment]
Verfied as well with next.261, looks good.

> Please avoid closing issues without verification.

Yes, please don't close issues that I open before I (myself) verified them. I'm happy to do the verification. 
I was just out of office for a bit this time.

**Comment by ethanbdev** [comment]
@eriklharper I am seeing that with `step="any"` a decimal also cannot be inputted. This doesn't seem right, what is intended?

https://codepen.io/mpayson/pen/XWRxXBr

**Comment by AdelheidF** [comment]
We can't upgrade to beta.61 because of this.

**Comment by babakbolour** [comment]
@AdelheidF just surfing the repo and saw this one!  the screenshot at the beginning comment shows RGB values, but I assume that  is just one application of the component. Can Calcite-input accept numbers bigger than 999? and if yes, then what is the behavior of this component, when an international user enters a number like this: "12 345" or "12'345"? in other words does the component accept different 'thousand separators symbols'? and hope there is not crash, if they do enter such number.

**Comment by AdelheidF** [comment]
> does the component accept different 'thousand separators symbols'?

yes, we tested that

**Comment by eriklharper** [comment]
@babakbolour correct.  The input component accepts the locale-appropriate thousand separators.

**Comment by eriklharper** [comment]
> @eriklharper I am seeing that with `step="any"` a decimal also cannot be inputted. This doesn't seem right, what is intended?
> 
> https://codepen.io/mpayson/pen/XWRxXBr

Definitely a bug.  The `any` step should allow all types of number values, including decimals according to the HTML spec.  https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/step  I'll create a bug issue to address this.

**Comment by AdelheidF** [comment]
Closing this. The new issue is handled via https://github.com/Esri/calcite-components/issues/2764
